lighthouse-plots/measure.js

```diff
@@ -0,0 +1,185 @@
+const parseURL = require('url').parse;
+
+const mkdirp = require('mkdirp');
+
+const utils = require('./utils.js');
+const config = require('./config/perf-metrics.json');
+
+const Lighthouse = require('../lighthouse-core/index.js');
+const ChromeLauncher = require('../lighthouse-cli/chrome-launcher.js').ChromeLauncher;
+const Printer = require('../lighthouse-cli/printer');
+const assetSaver = require('../lighthouse-core/lib/asset-saver.js');
+
+const OUT_PATH = path.resolve(__dirname, 'out');
+
+// TODO(chenwilliam): make this a flag (e.g. -n 10)
+const NUMBER_OF_RUNS = 20;
+
+const URLS = [
+  // Flagship sites
+  'https://nytimes.com',
+  'https://flipkart.com',
+  'http://www.espn.com/',
+  'https://www.washingtonpost.com/pwa/',
+
+  // Sourced from: https://en.wikipedia.org/wiki/List_of_most_popular_websites
+  // (http://www.alexa.com/topsites)
+  // Removed adult websites and duplicates (e.g. google int'l websites)
+  // Also removed sites that don't have significant index pages:
+  // "t.co", "popads.net", "onclickads.net", "microsoftonline.com", "onclckds.com", "cnzz.com",
+  // "live.com", "adf.ly", "googleusercontent.com",
+  'https://google.com',
+  'https://youtube.com',
+  'https://facebook.com',
+  'https://baidu.com',
+  'https://wikipedia.org',
+  'https://yahoo.com',
+  'https://amazon.com',
+  'http://www.qq.com/',
+  'https://taobao.com',
+  'https://vk.com',
+  'https://twitter.com',
+  'https://instagram.com',
+  'http://www.hao123.cn/',
+  'http://www.sohu.com/',
+  'https://sina.com.cn',
+  'https://reddit.com',
+  'https://linkedin.com',
+  'https://tmall.com',
+  'https://weibo.com',
+  'https://360.cn',
+  'https://yandex.ru',
+  'https://ebay.com',
+  'https://bing.com',
+  'https://msn.com',
+  'https://www.sogou.com/',
+  'https://wordpress.com',
+  'https://microsoft.com',
+  'https://tumblr.com',
+  'https://aliexpress.com',
+  'https://blogspot.com',
+  'https://netflix.com',
+  'https://ok.ru',
+  'https://stackoverflow.com',
+  'https://imgur.com',
+  'https://apple.com',
+  'http://www.naver.com/',
+  'https://mail.ru',
+  'http://www.imdb.com/',
+  'https://office.com',
+  'https://github.com',
+  'https://pinterest.com',
+  'https://paypal.com',
+  'http://www.tianya.cn/',
+  'https://diply.com',
+  'https://twitch.tv',
+  'https://adobe.com',
+  'https://wikia.com',
+  'https://coccoc.com',
+  'https://so.com',
+  'https://fc2.com',
+  'https://www.pixnet.net/',
+  'https://dropbox.com',
+  'https://zhihu.com',
+  'https://whatsapp.com',
+  'https://alibaba.com',
+  'https://ask.com',
+  'https://bbc.com'
+];
+
+async function main() {
+  if (utils.isDir(OUT_PATH)) {
+    console.log('Please remove or move ./out/');
+    return;
+  }
+
+  let launcher;
+  try {
+    launcher = new ChromeLauncher({ port: 9222, autoSelectChrome: true });
+    try {
+      await launcher.isDebuggerReady();
+    } catch (err) {
+      await launcher.run();
+    }
+    await runAnalysis();
```

Comment:
This is something i'd actually like to A/B test with this framework. (To see how much of a difference it makes). DNS costs will definitely be in relaunched ones.

In this model, it favors quicker collection, but also aligns closer with how people run LH with the extension and in devtools. So I think it's slightly better to debug that scenario first.